test_conformance/conversions/basic_test_conversions.cpp

```diff
@@ -282,15 +288,15 @@ static inline void Force64BitFPUPrecision(void)
 #endif
 }
 
-
-template <typename InType, typename OutType>
-int CalcRefValsPat<InType, OutType>::check_result(void *test, uint32_t count,
-                                                  int vectorSize)
+template <typename InType, typename OutType, bool InFP, bool OutFP>
+int CalcRefValsPat<InType, OutType, InFP, OutFP>::check_result(void *test,
+                                                               uint32_t count,
+                                                               int vectorSize)
 {
     const cl_uchar *a = (const cl_uchar *)gAllowZ;
 
-    if (std::is_integral<OutType>::value)
-    { // char/uchar/short/ushort/int/uint/long/ulong
+    if (std::is_integral<OutType>::value || is_half<OutType, OutFP>())
+    { // char/uchar/short/ushort/half/int/uint/long/ulong
```

Comment:
I don't think this will work - the half case needs to allow NaNs to be binary different, whereas the integral types do not have this special case.

Reply:
Done